docassemble/AssemblyLine/data/questions/al_saved_sessions.yml

```
@@ -0,0 +1,264 @@
+---
+template: al_sessions_preview_variables
+subject: |
+  Preview
+content: |
+  <pre>
+  ${ verbatim(json.dumps(al_simple_filtered_vars, sort_keys=True, indent=4)) }
+  </pre>  
+---
+id: save answer snapshot
+# Save a copy
+comment: |
+  Copy the answers from the "current session" into the 
+  "al_saved_session_store.yml" interview.
+question: |
+  Save answer snapshot
+subquestion: |
+  You have visited ${ all_variables(include_internal=True).get('_internal',{}).get('steps',1) }
+  screens so far and defined ${ len(al_sessions_filtered_vars) } fields.
+  
+  ${ collapse_template(al_sessions_preview_variables) }
+
+  Save a copy of your current answer progress, including any files, to load into
+  this interview or a different interview later.
+  
+  **Warning**: this feature is for developers only. A snapshot is not a replacement
+  for starting the interview from the beginning. Some features may not work the 
+  same when loaded from a snapshot.
+fields:
+  - Label: al_sessions_snapshot_label
+    default: |
+      ${ all_variables(special='titles').get('full') }
+      % if all_variables(special='titles').get('sub'):
+      : ${ all_variables(special='titles').get('sub') }
+      % endif
+      % if defined('users[0].name.first'):
+      - ${ comma_and_list(users.complete_elements()) }
+      % endif
+      % if defined('other_parties[0].name.first'):
+      - ${ comma_and_list(other_parties.complete_elements()) }
+      % endif
+    help: |
+      This can help you identify the snapshot later.
+      Use a cue like "10 children" or "No adverse party"
+---
+id: fast forward answers
+# fast forward
+comment: |
+  Copy the answers from any saved session into the current interview filename,
+  by creating a new session of the interview and copying answers into it.
+question: |
+  Fast Forward
+subquestion: |
+  Choose an interview below to load into a copy of the currently running
+  interview (${ all_variables(special='titles').get('full') }).
+  
+  **Warning**: this feature is for developers only. A snapshot is not a replacement
+  for starting the interview from the beginning. Some features may not work the 
+  same when loaded from a snapshot into a new interview.
+fields:
+  - Copy my answers from: al_sessions_source_session
+    datatype: combobox
+    code: al_formatted_sessions  
+under: |
+  ${ action_button_html(url_of('interviews'),label="Manage saved answers") }
+---
+code: |
+  sessions_list_temp = []
+  next_id = None
+  while True:
+      (items, next_id) = interview_list(next_id=next_id)
+      sessions_list_temp.extend(items)
+      if not next_id:
+          break
+  al_sessions_list = sessions_list_temp
+  del sessions_list_temp
+---
+code: |
+  formatted_sessions_temp = []
+  for index, session in enumerate(al_sessions_list):
+    formatted_sessions_temp.append({
+      index: f"{session.get('title')}:{session.get('subtitle','') if session.get('subtitle','') else ''} ({ format_date(session.get('modtime')) })"
```

When does `session.get('subtitle', '')` return something that's false, and not just a NoneException or something similar. At a glance it's redundant to the default attribute.

---

It's sometimes not present in the session information, and if it's not I just want to show the title and no `:`

Both `None` and `''` are Falsy. I think the second session.get('subtitle') does not need to have the default return though--that is potentially confusing. Good catch.

---

Ah, I think you want this instead: 
```suggestion
      index: f"{session.get('title'):session.get('subtitle') if session.get('subtitle', '') else session.get('title')} ({ format_date(session.get('modtime')) })"
```

---

no, the : needs to be outside of the `{}`. but i'll take a look and simplify it a bit.